packages/dev/gui/src/2D/controls/inputTextArea.ts

```diff
@@ -0,0 +1,1257 @@
+    }
+
+    /**
+     * Gets or sets outlineColor of the text to display
+     */
+    public get outlineColor(): string {
+        return this._outlineColor;
+    }
+
+    /**
+     * Gets or sets outlineColor of the text to display
+     */
+    public set outlineColor(value: string) {
+        if (this._outlineColor === value) {
+            return;
+        }
+        this._outlineColor = value;
+        this._markAsDirty();
+    }
+
+    /** Gets or sets a boolean indicating if the control can auto stretch its height to adapt to the text */
+    @serialize()
+    public get autoStretchHeight(): boolean {
+        return this._autoStretchHeight;
+    }
+
+    public set autoStretchHeight(value: boolean) {
+        if (this._autoStretchHeight === value) {
+            return;
+        }
+
+        this._autoStretchHeight = value;
+        this._markAsDirty();
+    }
+
+    public set height(value: string | number) {
+        this._fixedRatioMasterIsWidth = false;
+
+        if (this._height.toString(this._host) === value) {
+            return;
+        }
+
+        if (this._height.fromString(value)) {
+            this._markAsDirty();
+        }
+
+        this._autoStretchHeight = false;
+    }
+
+    @serialize()
+    public get maxHeight(): string | number {
+        return this._maxHeight.toString(this._host);
+    }
+
+    /** Gets the maximum width allowed by the control in pixels */
+    public get maxHeightInPixels(): number {
+        return this._maxHeight.getValueInPixel(this._host, this._cachedParentMeasure.height);
+    }
+
+    public set maxHeight(value: string | number) {
+        if (this._maxHeight.toString(this._host) === value) {
+            return;
+        }
+
+        if (this._maxHeight.fromString(value)) {
+            this._markAsDirty();
+        }
+    }
+
+    /**
+     * Creates a new InputTextArea
+     * @param name defines the control name
+     * @param text defines the text of the control
+     */
+    constructor(public name?: string, text: string = "") {
+        super(name);
+
+        this.text = text;
+
+        this.isPointerBlocker = true;
+
+        this.onLinesReadyObservable.add(() => this._updateCursorPosition());
+
+        this._highlightCursorInfo = {
+            initialStartIndex: -1,
+            initialRelativeStartIndex: -1,
+            initialLineIndex: -1,
+        };
+
+        this._cursorInfo = {
+            globalStartIndex: -1,
+            globalEndIndex: -1,
+            relativeEndIndex: -1,
+            relativeStartIndex: -1,
+            currentLineIndex: 0,
+        };
+    }
+
+    protected _getTypeName(): string {
+        return "InputTextArea";
+    }
+
+    /**
+     * Process the last keyboard input
+     *
+     * @param keyCode The ascii input number
+     * @param key The key string representation
+     * @param evt The keyboard event emits with input
+     * @hidden
+     */
+    public processKey(keyCode: number, key?: string, evt?: IKeyboardEvent) {

```

Comment:
keyCode was already deprecated and will slowly be removed from browsers. And we don't support IE11 anymore (bless whoever's in charge of that :-)) - https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Would it be possible to work with `key` or `code` instead? Or was there any specific reason for choosing to work with keyCode?

I understand we use this internally quite often. And we need to change that (sooner rather than later). I am just trying to avoid commiting new code with deprecated APIs.

Reply:
Hmm, I didn't notice that... I don't have any reason to keep this deprecated API. I will rewrite the processkey ASAP 👍

Reply:
@RaananW, unfortnaly I can't reproduce a correct behavior without the deprecation. The charCode filters the character and discard all modifier key like `Shift`, `Alt`, etc.

Actually, I can achieve almost the same things but I'm stuck to allow two duet characters: `^^` and ` `` `

Reply:
@RaananW, Can I push this evol?

Reply:
Can this be resolved, or do we have some more to discuss?

Reply:
We can resolve of course! :)